osu.Game/Skinning/SkinManager.cs

```diff
@@ -253,14 +253,10 @@ public IEnumerable<ISkin> AllSources
                 yield return CurrentSkin.Value;
 
                 // Skin manager provides default fallbacks.
-                // This handles cases where a user skin doesn't have the required resources for complete display of
-                // certain elements.
-
-                if (CurrentSkin.Value is LegacySkin && CurrentSkin.Value != DefaultClassicSkin)
-                    yield return DefaultClassicSkin;
-
-                if (CurrentSkin.Value != trianglesSkin)
-                    yield return trianglesSkin;
+                // This handles cases where a user skin doesn't have the required resources for complete display of certain elements, ie. textures.
+                if (CurrentSkin.Value != DefaultClassicSkin) yield return DefaultClassicSkin;
```

Comment:
I've just added all skins here for now, but it's arguable if we need them all. None of these skins modify resource lookups so one might be enough.